libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx

```diff
@@ -172,85 +187,100 @@ export const AttributeFilterButtonCore: React.FC<IAttributeFilterButtonProps> =
         "It's not possible to combine 'filter' property with 'connectToPlaceholder' property. Either provide a value, or a placeholder.",
     );
 
+    const [placeholder, setPlaceholder] = usePlaceholder(props.connectToPlaceholder);
+
     const getInitialSelectedOptions = (): IAttributeElement[] =>
         // the as any cast is ok here, the data will get fixed once the element load completes
         // this serves only to have some initial state here so that when full element data is loaded
         // it automatically sets the props.filter.elements as selected
-        props.filter
-            ? (attributeElementsToAttributeElementArray(filterAttributeElements(props.filter)) as any)
+        placeholder || props.filter
+            ? (attributeElementsToAttributeElementArray(
+                  filterAttributeElements(placeholder || props.filter),
+              ) as any)
             : [];
+
     const getInitialIsInverted = (): boolean =>
-        props.filter ? isNegativeAttributeFilter(props.filter) : true;
-    const [validElements, setValidElements] = useState<IElementQueryResultWithEmptyItems>(null);
-    const [selectedFilterOptions, setSelectedFilterOptions] = useState(getInitialSelectedOptions);
-    const [isInverted, setIsInverted] = useState(getInitialIsInverted);
-    const [prevSelectedFilterOptions, setPrevSelectedFilterOptions] = useState(getInitialSelectedOptions);
-    const [prevIsInverted, setPrevIsInverted] = useState(getInitialIsInverted);
-    const [firstLoad, setFirstLoad] = useState(true);
-    const [isLoadingElements, setIsLoadingElements] = useState(true);
-    const [error, setError] = useState<any>(null);
-    const [searchString, setSearchString] = useState("");
-    const [offset, setOffset] = useState(0);
-    const [limit, setLimit] = useState(LIMIT);
-    const [totalCount, setTotalCount] = useState(LIMIT);
-    const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+        placeholder || props.filter ? isNegativeAttributeFilter(placeholder || props.filter) : true;
+
+    const [state, setState] = useState<IAttributeFilterButtonState>({
```

Comment:
Nitpick: please use lazy state init like 
```ts
const [state, setState] = useState<IAttributeFilterButtonState>(() => ({
   ...
}));
```

That way the `getInitialSelectedOptions` will not be called on subsequent renders of the components just to be ignored (because the initialState is not used on subsequent renders, but the object is constructed every time unless using the lazy state init).

Reply:
Also, you could only compute the `getInitialSelectedOptions` and `getInitialIsInverted` once, not twice, using a local variable in the initializer function.